src/lib.rs

```diff
@@ -75,9 +77,32 @@ pub fn sealed(_args: TokenStream, input: TokenStream) -> TokenStream {
     })
 }
 
+// generously offered by @danielhenrymantilla aka Yandros
+fn to_snake_case(s: &'_ str) -> String {
+    let mut ret = String::with_capacity(s.len());
+    let mut first = true;
+    s.bytes().for_each(|c| {
+        if c.is_ascii_uppercase() {
+            if !first {
+                ret.push('_');
+            }
+            ret.push(c.to_ascii_lowercase() as char);
+        } else {
+            ret.push(c as char);
+        }
+        first = false;
+    });
+    ret
+}
+
+fn seal_name<D: Display>(seal: D) -> syn::Ident {
+    ::quote::format_ident!("__seal_for_{}", to_snake_case(&seal.to_string()),)
```

Comment:
As for me, a `__seal_` prefix would be enough. Keep in mind, that these names will appear in `cargo doc` generated documentation, so making them too long/complicated may be not reading-friendly.

Reply:
FWIW, this name will only appear with `document-private-items`, though, since the module is `pub(crate)`; so the actual name does not matter that much, imho. It may actually matter when error messages happen, though (_e.g._, somebody forgetting to write `#[sealed]` on the `impl`), and that's thus the main case where readability comes into play. At which point it is subjective: I don't have any particular preference myself.

Reply:
@danielhenrymantilla as I recall, we will see them module path in trait bounds in documentation, it won't be clickable though. You may check it by running `cargo doc`.

Reply:
I shortened the name, changing it shouldn't be hard and in the future I think adding a parameter to rename the seal can be a decent approach for such cases